internal/integration/crud_helpers_test.go

```diff
@@ -1638,7 +1648,11 @@ func verifyCursorResult(mt *mtest.T, cur *mongo.Cursor, result interface{}) {
 	}
 }
 
-func verifySingleResult(mt *mtest.T, actualResult *mongo.SingleResult, expectedResult interface{}) {
+func verifySingleResult[T bson.Raw | bson.RawArray](
+	mt *mtest.T,
+	actualResult *mongo.SingleResult[T],
+	expectedResult interface{},
+) {
```

Comment:
`verifySingleResult` is only ever called with a `bson.Raw`, so the function doesn't need to be parameterized. Then the `bson.Raw` conversion in the call to `compareDocs` can also be removed.

E.g. function signature:
```go
func verifySingleResult(
	mt *mtest.T,
	actualResult *mongo.SingleResult[bson.Raw],
	expectedResult interface{},
) {
```